casepro/backend/junebug.py

```diff
@@ -463,3 +468,53 @@ def received_junebug_message(request):
         created_on=datetime.now(), has_labels=True)
 
     return JsonResponse(msg.as_json())
+
+
+@csrf_exempt
+def receive_identity_store_optout(request):
+    '''Handles optout notifications from the Identity Store.'''
+    if request.method != 'POST':
+        return JsonResponse({'reason': 'Method not allowed.'}, status=405)
+
+    try:
+        data = json.loads(request.body)
+    except ValueError as e:
+        return JsonResponse(
+            {'reason': 'JSON decode error', 'details': e.message}, status=400)
+
+    try:
+        identity_id = data['identity']
+        optout_type = data['optout_type']
+    except KeyError as e:
+        return JsonResponse(
+            {'reason': 'Both "identity" and "optout_type" must be specified.'},
+            status=400)
+
+    syncer = IdentityStoreContactSyncer()
+
+    # The identity store currently doesn't specify the response format or do
+    # anything with the response.
+
+    # TODO: Support existance and usage of multiple orgs
+    org = Org.objects.first()
+    local_contact = syncer.fetch_local(org, identity_id)
+    if local_contact:
+        local_contact.lock(org, identity_id)
+        if optout_type == "forget":
+            local_contact.release()
```

Comment:
I'm wondering if we should leave a `TODO` here to remove the contact more fully. E.g. we should also remove any additional information we happen to have on the contact. One way to handle this would be to add a `.forget()` method to the contact model. For now it can just do `.release()` but but in future it might do more. That should also at least get the CasePro maintainers attention on the question of whether there is still more to do.